validator/src/journal/block_manager.rs

```diff
@@ -398,13 +402,158 @@ impl BlockManagerState {
 #[derive(Default, Clone)]
 pub struct BlockManager {
     state: Arc<BlockManagerState>,
+    persist_lock: Arc<RwLock<()>>,
 }
 
 impl BlockManager {
     pub fn new() -> Self {
         BlockManager::default()
     }
 
+    fn transaction_index_contains(
+        &self,
+        index: &Box<IndexedBlockStore>,
+        block_id: &str,
+        id: &str,
+    ) -> Result<bool, BlockManagerError> {
+        let index = index.as_ref();
+        if TransactionIndex::contains(index, id)? {
```

Comment:
I think if you make this function take `&TransactionIndex` that problem goes away.

Reply:
Yeah, I think that is right. It could be &TransactionIndex and &BatchIndex for the signature. I would like to make this function generic on which index is being used. That would be a win in reducing code.